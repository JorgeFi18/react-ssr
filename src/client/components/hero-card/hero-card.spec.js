import React from 'react';
import renderer from 'react-test-renderer';
import HeroCard from './hero-card.component';

const mockHero = {
    name: 'Drax',
    thumbnail: 'path/to/img',
    modified:'2014-04-29T14:18:17-0400',
    description: ''
};

describe('Hero Card component', () => {

    it('Name should be Drax', () => {
        const component = renderer.create(
            <HeroCard {...mockHero} />
        )

        const tree = component.toJSON();
        const [ , foreground ] = tree.children;
        const [ , nameWrapper ] = foreground.children;
        const name = nameWrapper.children[0];

        expect(name).toBe('Drax');
    });

    it('Should match with snapshot', () => {
        const component = renderer.create(
            <HeroCard {...mockHero} />
        )

        const tree = component.toJSON();
        const [ , foreground ] = tree.children;
        const [ , , dateWrapper ] = foreground.children;
        const date = dateWrapper.children[0];

        expect(date).toMatchSnapshot();
    });
});