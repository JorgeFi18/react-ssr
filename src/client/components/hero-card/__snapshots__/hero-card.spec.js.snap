// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Card component Should match with snapshot 1`] = `"April 29th 2014"`;
